File: shared/types/meeting.ts
export interface Meeting {
  id: string;
  title: string;
  description?: string;
  file_path: string;
  file_name: string;
  file_size: number;
  duration?: number;
  transcription?: string;
  summary?: string;
  status: MeetingStatus;
  user_id: string;
  created_at: string;
  updated_at: string;
  tasks?: Task[];
}

export enum MeetingStatus {
  UPLOADING = "uploading",
  PROCESSING = "processing",
  TRANSCRIBING = "transcribing",
  SUMMARIZING = "summarizing",
  COMPLETED = "completed",
  FAILED = "failed"
}

export interface MeetingCreate {
  title: string;
  description?: string;
  file: File;
}

export interface MeetingUpdate {
  title?: string;
  description?: string;
}

export interface MeetingUpload {
  meeting_id: string;
  upload_url: string;
}

File: backend/app/models/meeting.py
from sqlalchemy import Column, Integer, String, DateTime, Text, Float, ForeignKey, Enum
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
import enum
from app.core.database import Base


class MeetingStatus(str, enum.Enum):
    UPLOADED = "uploaded"
    TRANSCRIBING = "transcribing"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"

class Meeting(Base):
    __tablename__ = "meetings"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False)
    description = Column(Text)
    file_path = Column(String, nullable=False)
    file_name = Column(String, nullable=False)
    file_size = Column(Integer)
    duration = Column(Float)  # in seconds
    status = Column(Enum(MeetingStatus), default=MeetingStatus.UPLOADED)
    
    # AI Processing Results
    transcription = Column(Text)
    summary = Column(Text)
    key_points = Column(Text)  # JSON string
    
    # Relationships
    owner_id = Column(Integer, ForeignKey("users.id"))
    owner = relationship("User", back_populates="meetings")
    tasks = relationship("Task", back_populates="meeting")
    
    # Timestamps
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    processed_at = Column(DateTime(timezone=True))


